Feature/gpu as a service

shebistar opened the pull request:
# What does this PR do?
Add GPU as a Service content and overlay for NVIDIA GPU with OCP 4.19 and RHOAI 2.22




## Test Plan

* Deploy new cluster
* Use bootstrap
* Select option for rhoai-stable-2.22-aws-gpu-time-sliced

strangiato commented on components/operators/openshift-ai/instance/components/model-server-pod-sizes/patch-rhoai-dashboard.yaml:
```diff
@@ -12,6 +12,14 @@ spec:
         requests:
           cpu: '1'
           memory: 4Gi
+    - name: Small-Medium
```
This style of resource configuration is depreciated with accelerator profiles.

Consider using hardware profiles instead:
https://docs.redhat.com/en/documentation/red_hat_openshift_ai_self-managed/2.22/html/working_with_accelerators/working-with-hardware-profiles_accelerators#creating-a-hardware-profile_accelerators

shebistar commented on components/operators/gpu-operator-certified/operator/components/console-plugin-25/nvidia-dcgm-exporter-dashboard-cm.yaml:
```diff
@@ -0,0 +1,831 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: nvidia-dcgm-exporter-dashboard
```
I don't see any difference, but I did not see it applied when I was applying the overlay, maybe I can include this monitoring-dashboard to this resource.

shebistar commented on components/operators/openshift-ai/instance/components/notebook-pod-sizes/patch-rhoai-dashboard.yaml:
```diff
@@ -20,6 +20,14 @@ spec:
         requests:
           cpu: '3'
           memory: 24Gi
+    - name: Small-Medium
```
I did not touch this file on my development, maybe it was there before, if you have a suggestion for deleting it or modifying it, please do it.

strangiato commented on components/operators/gpu-operator-certified/operator/base/kustomization.yaml:
```diff
@@ -7,4 +7,5 @@ resources:
 components:
-  - ../components/console-plugin
+  - ../components/console-plugin-25
+#  Console Plugin Deprecated using helm: see: https://docs.nvidia.com/datacenter/cloud-native/openshift/25.3.2/enable-gpu-monitoring-dashboard.html
```
I think there may be a little confusion here between the `NVIDIA Console Plugin` and the `NVIDIA Monitoring Dashboard`.

The link you provided includes details for configuring the Monitoring Dashboard which we already have code for here:
https://github.com/redhat-ai-services/ai-accelerator/blob/main/components/operators/gpu-operator-certified/instance/components/monitoring-dashboard/kustomization.yaml

Note that this is located in the `instance` folder because it is dependent on the DCGM exporter which doesn't get created until after the GPU instance is created.  Console plugin existed in the `operator` folder because generally `console-plugins` are generally shipped with the operator and need to be enabled with the subscription (this one was just never officially packaged with the operator).

The Console Plugin is based off of code that is available here:
https://github.com/rh-ecosystem-edge/console-plugin-nvidia-gpu/

I don't see anything that indicates that the console plugin is officially depreciated, but it doesn't look like this doc  it doesn't exist in 25.x anymore:
https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/openshift/enable-gpu-op-dashboard.html

The Console Plugin never really had a lot of traction by NVIDIA so I'm not surprised if it was quietly depreciated.  It has always been "unofficially" maintained by RH and it looks like we have provided some minor updates to it but if NVIDIA is removing references to it from their official docs we may want to pull it out entirely....

shebistar commented:
I like more the idea of using the  https://github.com/redhat-ai-services/ai-accelerator-examples/ to put the solution for the lab, we are trying to build an independent folder, not tied with the tenants folder used before. I have removed it from this repo and moved it with this PR: https://github.com/redhat-ai-services/ai-accelerator-examples/pull/10

About the time-slicing concepts, we were not able to get a GPU with MIG support from the demo team. We only got A10 GPU, and the only one supported for MIG in the AWS region we are currently working is the A100, which is too expensive, and we would need to build a business case around it. So we are explaining during the Bootcamp the different ways to have GPU partitioning, and that because of lack of resources in our environment we will test only with time-slice. Our idea initially was to go directly with MIG, because it is what makes more sense for production use cases.
